src/BloomExe/Edit/EditingModel.cs

```diff
@@ -496,8 +497,13 @@ void OnPageSelectionChanged(object sender, EventArgs e)
 			{
 				if (_previouslySelectedPage != null && _domForCurrentPage != null)
 				{
-					if(!_inProcessOfDeleting)//this is a mess.. before if you did a delete and quickly selected another page, events transpired such that you're now trying to save a deleted page
+					// Before if you did a delete and quickly selected another page,
+					// events transpired such that you're now trying to save a deleted page
+					if (!_inProcessOfDeleting)
+					{
+						CheckForUnsavedOrigamiAndSave();
```

Comment:
Why isn't this called from SaveNow()? Are there cases where we want to save the page but do NOT need to deal with unsaved origami?